src/primitives/transaction.cpp

```diff
@@ -43,6 +43,40 @@ std::string CTxIn::ToString() const
     return str;
 }
 
+
+bool CTxIn::SpendsNestedPayToWitnessPubKeyHashOutput(CScript spentScriptPubKey) const {
+    if ((!spentScriptPubKey.IsPayToScriptHash()) || this->scriptWitness.IsNull()) {
```

Comment:
I don't think you need the `this->` pointer derefences here. Just `scriptWitness.IsNull()` should be fine.

(same for `this->` repetitions below)